work/worker.go

```diff
@@ -500,11 +500,11 @@ func (self *worker) commitNewWork() {
 	tstart := time.Now()
 	tstamp := tstart.Unix()
 	if self.nodetype == common.CONSENSUSNODE {
-		ideal := parent.Time().Int64() + params.BlockGenerationInterval
+		ideal := time.Unix(parent.Time().Int64()+params.BlockGenerationInterval, 0)
```

Comment:
If you also use `TimeFoS`, calculation would be more precise.
https://github.com/klaytn/klaytn/blob/58581aa1c43324a5d8a2fa67477bdd03ab921ffb/consensus/istanbul/backend/engine.go#L398-L402

Reply:
`ideal` is always underestimated unless we take `TimeFoS` into account, since `TimeFoS` is always a positive value. And we compare `tstart` and `ideal` to determine whether we take a sleep here or not, there is higher probability of taking a sleep when not reflecting `TimeFoS`.

- ideal w/o TimeFoS = 1000
- ideal w/   TimeFoS = 1100
- tstart                      = 1050

In this case, if we use TimeFoS, we can avoid sleeping but we need to take 50ms of sleep if we don't use `TimeFoS`. Please correct me if I misunderstood.

Reply:
@ehnuje  In your case, I think it doesn't sleep w/ TimeFoS because `start` is before `ideal w/o TimeFoS` 
I think underestimating `ideal` is better to keep maintaining 1 second block time. 
Do I miss something?

Reply:
> 

I also agree with Aidan's comment. I expect the block miner should wait as little as possible as well as the two different blocks have different timestamp. I believe removing the fraction of seconds can help reduce the waiting time. Please let me know if my idea is incorrect.